src/test/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs

```diff
@@ -160,6 +170,28 @@ public void ItFailsToSetGUISubsystemWithWrongDefault()
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void ItGeneratesExecutableImage()
+        {
+            using (TestDirectory testDirectory = TestDirectory.Create())
+            {
+                string sourceAppHostMock = PrepareAppHostMockFile(testDirectory);
```

Comment:
For the test to be effective the file needs to start with the "wrong" permissions - ideally read-only to everybody.

Reply:
As noted in https://github.com/dotnet/core-setup/pull/8510#issuecomment-540761524 `SearchAndReplace` was throwing exception due to readonly-ness, with octal value `444` (readonly for everyone: `-r--r--r--`). By setting permissions right after FileCopy in HostWriter as per your comment above, it was also made possible. #fixed